pkg/ovs/ovsctl/ofctl.go

```diff
@@ -64,11 +63,11 @@ func (c *ovsCtlClient) DumpTableFlows(table uint8) ([]string, error) {
 	return c.DumpFlows(fmt.Sprintf("table=%d", table))
 }
 
-func (c *ovsCtlClient) DumpGroup(groupID int) (string, error) {
+func (c *ovsCtlClient) DumpGroup(groupID uint32) (string, error) {
 	// There seems a bug in ovs-ofctl that dump-groups always returns all
 	// the groups when using Openflow13, even when the group ID is provided.
 	// As a workaround, we do not specify Openflow13 to run the command.
```

Comment:
Unrelated to this PR as well, but this comment looks incorrect. As per the documentation for `dump-groups`:

> Only  OpenFlow  1.5  and later support dumping a specific group.
              Earlier versions of OpenFlow always dump all groups.

When we omit the OpenFlow version, it will default to OpenFlow10, _but also enable the Nicira extensions apparently, which support dumping a single group._ When specifying OpenFlow13 explicitly, it will not enable these extensions and as per the manual it does not support dumping a single group. As for specifying OpenFlow15, we do not enable this protocol on the bridge at the moment. So there is no bug in ovs-ofctl.

Reply:
Thanks for figuring this out! Updated the comments.
Still think ovs-ofctl should at least provide some hints when a group ID is specified with a OpenFlow version does not support that.